app/controllers/application_controller.rb

```diff
@@ -54,4 +54,14 @@ def require_authenticated_user
       redirect_to main_app.user_omniauth_authorize_path(:shibboleth)
     end
   end
+
+  def authorize_report_access
```

Comment:
Actually this was fixed as you can see below it uses the can? method. Whereas before I was manually checking roles. But you're right the `authorize_report_access` method must exist.